src/qmapshack/poi/CPoiDraw.cpp

```diff
@@ -137,6 +138,36 @@ void CPoiDraw::loadPoiPath(QSettings &cfg)
     poiPaths = cfg.value("poiPaths", poiPaths).toStringList();
 }
 
+poi_t CPoiDraw::findPOICloseBy(const QPoint &px) const
+{
+    poi_t poi;
+    CPoiItem::mutexActivePois.lock();
+    if(poiList)
+    {
+        for(int i = 0; i < poiList->count(); i++)
+        {
+            auto item = poiList->item(i);
+
+            if(!item || item->getPoifile().isNull())
+            {
+                // as all active maps have to be at the top of the list
+                // it is ok to break as soon as the first map with no
+                // active files is hit.
+                break;
+            }
+
+            item->getPoifile()->findPOICloseBy(px, poi);
+            if(poi.pos != NOPOINTF)
```

Comment:
`findPOICloseBy()` could have a boolean return value to tell if the search has been successful.